src/LuauAST/util/isValidNumberLiteral.ts

```diff
@@ -1,4 +1,4 @@
-const DECIMAL_LITERAL_REGEX = /^(?:\d[\d_]*(?:\.[\d_]*)?|\.\d[\d_]*)(?:[eE][\+\-]?_*\d[\d_]*)?$/;
+const DECIMAL_LITERAL_REGEX = /^(?:\d[\d_]*(?:\.[\d_]*)?|\.\d[\d_]*)(?:[eE][+-]?_*\d[\d_]*)?$/;
```

Comment:
Did ESLint recommend this change?

Reply:
Yes. + and - don't need to be escaped inside character sets, because they are always literal in there.